schedule.ghactivity.php

```diff
@@ -52,6 +62,78 @@ public function record_average_label_time( $repo_name, $label ) {
 			// update_term_meta( $term_taxonomy_id[0], 'issue_slugs', $record_slugs );
 		}
 	}
+
+	/**
+	 * Task for updating list of repo labels for all the monitored(full_reporting) repos.
+	 *
+	 * @since 2.0.0
+	 */
+	public function record_repo_labels() {
+		$api   = new GHActivity_GHApi();
+		$repos = GHActivity_Queries::get_monitored_repos();
+
+		foreach ( $repos as $term_id => $name ) {
+			$labels = $api->get_repo_label_names( $name );
+			update_term_meta( $term_id, 'repo_labels', $labels );
+		}
+	}
+
+	/**
+	 * Records current Repo Label State in format $label => $labeled_issue_count
+	 * Splits all the labels in categories mentioned between `[` & `]`.
+	 * Example categories for Jetpack repo: Pri, Status, Type, none
+	 *
+	 * @since 2.0.0
+	 */
+	public function record_current_repo_labels_state() {
+		$repos = GHActivity_Queries::get_monitored_repos();
+
+		foreach ( $repos as $term_id => $repo_name ) {
+			$repo_labels_state = GHActivity_Queries::current_repo_labels_state( $repo_name );
+			$final_state       = array();
+			foreach ( $repo_labels_state as $label => $issue_slugs ) {
+				if ( strpos( $label, ']' ) !== false ) {
```

Comment:
Somehow this makes me wonder if we should update the labels in the Jetpack repo to prefix all labels about a specific module. 

This way we could separate those from things like "Internationalization" or "Performance". We could add something like "[Module]" or "[Feature]" in front of all module labels.

Reply:
That makes a lot of sense to me.
`[Feature] Sitemaps`
`[Meta] Docs` (like Internationalization too)
etc

Reply:
Yeah, I like that!